### DISCUSSION TITLE
Second order time derivatives (no elasticity)!

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#27718] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello,

I have searched quite a bit and cannot find any simple example of adding a time kernel of d^2u/dt^2 where u is an arbitrary `Variable`. I seem to remember it was just sitting in framework/src/kernels or maybe in tensor mechanics many years ago. 

I found numerous examples of Newmark and CentralDifference dynamics in the solid mechanics.  From what I understand, all of the dynamic elastic problems compute Newmark vel and accel variables and then provide them to the `InertialForce` Kernel which is linear in the acceleration of the displacements. 

Is this the procedure now in general for a non-elastic variable? 

First compute its Newmark acceleration, provide a Kernel linear in the acceleration, and then choose Newmark or CentralDiff time integration? Is it possible just to use InertialForce here (say variable = u)? There are no examples of this in the docs or tests so I have to ask. I would go and try this before posting but I am away from the MOOSEing equipment at the moment.

I did find `VectorSecondTimeDerivative` in the EM module but that is for a `VectorVariable` which I do not have and rather confused if that also can just accept regular Variables instead. I do know that all of the dotdot_u exist in the coupleable interface but just thought I would check if there is something already ready to go

Please advise! Thanks!
John

**Comment by mangerij** [comment]
I answer my own question :)

```
SecondTimeDerivativeScaled::SecondTimeDerivativeScaled(const InputParameters & parameters) :
    TimeKernel(parameters),
    _time_scale(getParam<Real>("time_scale")),
    _u_dot_dot(dotDot()),
    _du_dot_dot_du(dotDotDu())
{
 uDotDotRequested() = true;
}

Real
SecondTimeDerivativeScaled::computeQpResidual()
{
  return _time_scale * _test[_i][_qp] * _u_dot_dot[_qp];
}

Real
SecondTimeDerivativeScaled::computeQpJacobian()
{
  return _time_scale * _test[_i][_qp]*_phi[_j][_qp] * _du_dot_dot_du[_qp];
}

```

Still I am unsure why this is not a default Kernel in the framework? Is there something I am missing ?

**Reply by GiudGiud** [reply]
No idea. I'd venture it was only used by the mechanics folks who introduced their own time integrator and did all the work there

We can add it. Thanks for sharing the code. Feel free to make a pull request if you want credit for it (as a minor feature, see authorship guidelines)

**Reply by mangerij** [reply]
Quite funny that this compiles with MOOSE cloned two days ago and yet a fresh compile today it does not. 

```
/home/ferret/src/kernels/SecondTimeDerivativeScaled.C:43:2: error: 'uDotDotRequested' was not declared in this scope
   43 |  uDotDotRequested() = true;
```

Removing this line and it compiles and runs just fine.

**Reply by GiudGiud** [reply]
I think this is a routine on the problem?
Not something in the transientInterface
